torch/CMakeLists.txt

```diff
@@ -356,6 +356,11 @@ if(USE_UCC)
   target_compile_definitions(torch_python PRIVATE USE_UCC)
 endif()
 
+# Allow torch_python to link to OpenTelemetry API headers
+set(OPENTELEMETRY_API_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../third_party/opentelemetry-cpp/api/include")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../third_party/opentelemetry-cpp/api")
+target_link_libraries(torch_python PRIVATE opentelemetry_api)
+
```

Comment:
I guess this should be conditional on some feature, isn't it? Or should it always be enabled for all PyTorch configs?

Reply:
I don't want to have to ifdef opentel uses, so we want to unconditionally depend on the headers. I was promised that opentel will actually link against a no-op implementation so this won't result in any actual runtime dependency increase, but the details here matter.